ambry-store/src/main/java/com.github.ambry.store/HardDeleter.java

```diff
@@ -104,24 +109,33 @@
     throttler = new Throttler(config.storeHardDeleteBytesPerSec, 10, true, time);
     scanSizeInBytes = config.storeHardDeleteBytesPerSec * 10;
     messageRetentionSeconds = config.storeDeletedMessageRetentionDays * Time.SecsPerDay;
+    hardDeleterSleepTimeWhenCaughtUpMs = config.storeHardDeleterSleepTimeWhenCaughtUpInMs;
   }
 
   @Override
   public void run() {
     try {
       while (running.get()) {
```

Comment:
yes, I did try few options. I couldn't get it simpler. But to some extent the block in lines 121 to 127 is to handle paused. And lines from 128 to 138 if not for paused. Do you have any other better ideas.